build-logic/documentation/src/main/groovy/gradlebuild/docs/GradleBuildDocumentationPlugin.java

```diff
@@ -73,10 +75,15 @@ private void applyConventions(Project project, TaskContainer tasks, ObjectFactor
             // Javadocs reference goes into javadoc/
             task.from(extension.getJavadocs().getRenderedDocumentation(), sub -> sub.into("javadoc"));
 
+            // Dokka Kotlin DSL reference goes into dokka/
+            task.from(extension.getDokkadocs().getRenderedDocumentation(), sub -> sub.into("dokka"));
```

Comment:
Let's not name the folder and thus URL path fragment `dokka` but `kotlin-dsl`  instead.